xwiki-platform-core/xwiki-platform-scheduler/xwiki-platform-scheduler-ui/src/main/resources/XWiki/SchedulerJobSheet.xml

```diff
@@ -39,48 +39,61 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
-  <content>{{velocity}}
-#if($doc.getObject('XWiki.SchedulerJobClass'))
-#set($job = $doc.getObject("XWiki.SchedulerJobClass"))
-**$services.localization.render('xe.scheduler.job.name')**
-
-$!doc.display('jobName')
-
-**$services.localization.render('xe.scheduler.job.description')**
-
-$!doc.display('jobDescription')
-
-**$services.localization.render('xe.scheduler.job.expression')**
-
-$!doc.display('cron')
-
-**$services.localization.render('xe.scheduler.job.script')**
-
-#if($xcontext.action=="edit")
-$!doc.display("script")
-#elseif($xcontext.action=="view" &amp;&amp; "$!job.getProperty('script').value" !="")
-{{code language="groovy"}}
-$job.getProperty('script').value
-{{/code}}
+  <content>{{velocity output='false'}}
+#macro (displayJobProperty $obj $propName)
+  #set ($class = $obj.xWikiClass)
+  #set ($prettyName = $class.get($propName).translatedPrettyName)
+  &lt;dt&gt;
+    &lt;label for="${class.getName()}_${obj.number}_${propName}"&gt;$prettyName&lt;/label&gt;
+  &lt;/dt&gt;
+  &lt;dd&gt;
+    $!doc.display($propName, $obj)
+  &lt;/dd&gt;
 #end
+{{/velocity}}
 
-#if($xcontext.action=="edit")
-{{info}}$services.localization.render('xe.scheduler.job.scriptexplanation'){{/info}}
+{{velocity}}
+#if ($doc.getObject('XWiki.SchedulerJobClass'))
+  #set ($job = $doc.getObject('XWiki.SchedulerJobClass'))
+  {{html wiki='true'}}
+  &lt;div class = 'xform'&gt;
+    &lt;dl&gt;
+      #displayJobProperty($job 'jobName')
+      #displayJobProperty($job 'jobDescription')
+      #displayJobProperty($job 'cron')
+      #set ($class = $job.xWikiClass)
+      #set ($prettyName = $class.get('script').translatedPrettyName)
+      &lt;dt&gt;
+          &lt;label for="${class.getName()}_0_script"&gt;$prettyName&lt;/label&gt;
+          &lt;span class="xHint"&gt;
+            $services.localization.render('xe.scheduler.job.scriptexplanation')
+          &lt;/span&gt;
+      &lt;/dt&gt;
+      &lt;dd&gt;
+        #if ($xcontext.action == 'edit')
+          $!doc.display('script')
+        #else
+          ((({{code language='groovy'}}$!job.getValue('script'){{/code}})))
+        #end
+      &lt;/dd&gt;
```

Comment:
I do not like the code duplication done above. Instead, you should handle the special property `script` inside the `displayJobProperty` macro.

You should try to avoid duplication as much as possible, specially when it's about handling implementation details in more than one places.